website/docs/using-odo/json-output.md

```diff
@@ -0,0 +1,43 @@
+---
+title: JSON Output
+sidebar_position: 2
+---
+# JSON Output
+
+The `odo` commands that output some content generally accept a `-o json` flag to output this content in a JSON format, suitable for other programs to parse this output more easily.
+
+The output structure is similar to Kubernetes resources, with `Kind`, `apiVersion`, `metadata` ,`spec` and `status` fields.
+
+List commands return a `List` resource, containing an `items` (or similar) field listing the items of the list, each item being also similar to Kubernetes resources.
+
+Delete commands return a `Status` resource; see the [Status Kubernetes resource](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/status/).
+
+Other commands return a resource associated with the command (`Application`, `Storage`', `URL`, etc).
+
+The exhaustive list of commands accepting the `-o json` flag is currently:
+
+| commands                       | Kind (version)                          | Kind (version) of list items                                 | Complete content?         | 
+|--------------------------------|-----------------------------------------|--------------------------------------------------------------|---------------------------|
+| odo application describe       | Application (odo.dev/v1alpha1)          | *n/a*                                                        |no                         |
+| odo application list           | List (odo.dev/v1alpha1)                 | Application (odo.dev/v1alpha1)                               | ?                         |
+| odo catalog list components    | List (odo.dev/v1alpha1)                 | ComponentType (odo.dev/v1alpha1) (s2i) / *missing* (devfile) | yes (s2i) / yes (devfile) |
+| odo catalog list services      | List (odo.dev/v1alpha1)                 | ClusterServiceVersion (operators.coreos.com/v1alpha1)        | ?                         |
+| odo catalog describe component | *missing*                               | *n/a*                                                        | yes                       |
```

Comment:
How's "missing" different from "n/a"?

Reply:
"missing" indicates that the output does not contain the kind/version information, but should contain it.